coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java

```diff
@@ -556,6 +557,15 @@ private static Schema getUnionFieldSchema(@Nonnull Schema leftSchema, @Nonnull S
           if (leftSchema.getEnumSymbols().size() == rightSchema.getEnumSymbols().size()) {
             return leftSchema;
           }
+          // Check if the symbols of one Enum is a strict subset of the other
+          final ImmutableSet<String> leftSchemaSymbols = ImmutableSet.copyOf(leftSchema.getEnumSymbols());
+          final ImmutableSet<String> rightSchemaSymbols = ImmutableSet.copyOf(rightSchema.getEnumSymbols());
+          if (leftSchemaSymbols.containsAll(rightSchemaSymbols)) {
+            return leftSchema;
+          }
+          if (rightSchemaSymbols.containsAll(leftSchemaSymbols)) {
+            return rightSchema;
+          }
           break;
```

Comment:
Could you verify if we can union enum types like `[a, b]` and `[c, d]` in Hive?
If that is allowed, I think we may need to return the union of these enums (`[a, b, c, d]` for the above example)
If only subset-union is allowed, I think we need to remove 
```
          if (leftSchema.getEnumSymbols().size() == rightSchema.getEnumSymbols().size()) {
            return leftSchema;
          }
```

Reply:
Sounds good, will do. Thanks for pointing this out.